### PR TITLE
console: integration icon component and navigation bar layout fix (close #4187)

### LINKED ISSUE #4187: console: icon component integration
Integrate <code>Icon</code> component from the Design-System and replace all existing icons.

reference https://github.com/hasura/graphql-engine/issues/4136

### PR DESCRIPTION
**Pull request by lastnamearya**
### Description

This is the first Design-System integration PR.

----------------------------------------------------------------------------------------------

- Integration of the Icon component in console 
- 120 old icons replaced by the new icon component
- 42 new icons added in the Design-System
- Icon component's default size changed to <code>"14px"</code>
- New <code>pointer</code> prop for Icon component
- Header navigation bar layout styling fixed
- Some class components changed to functional where no state is used
  - BreadCrumb
  - LeftSubSidebar
  - Common
  - ApiCollectionPanel
  - HeadersEditor

### Changelog

- [ ] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [x] Console

### Related Issues

#4187 

#### Breaking changes

- [ ] No Breaking changes

### CODE REVIEW

File: console/src/components/Common/Layout/BreadCrumb/BreadCrumb.js [hunk 1]
```diff
@@ -2,48 +2,47 @@ import React from 'react';
...
-
-        const addArrow = () => [
-          <Sp key={'breadcrumb-space-before' + i} />,
-          <i key={'l' + i} className="fa fa-angle-right" aria-hidden="true" />,
-          <Sp key={'breadcrumb-space-after' + i} />,
-        ];
+import { Icon } from '../../../UIKit/atoms';
+import styles from '../../TableCommon/Table.scss';
+
+const BreadCrumb = ({ breadCrumbs }) => {
```
[the remark this answers is not in the document]
Comment (lastnamearya) [reply console/src/components/Common/Layout/BreadCrumb/BreadCrumb.js]: I think it's not going to be an issue cc @rikinsk

[the remark this answers is not in the document]
Comment (rikinsk) [reply console/src/components/Common/Layout/BreadCrumb/BreadCrumb.js]: Sure. This looks fine as well

File: console/src/components/Common/CustomInputTypes/JsonInput.js [hunk 2]
```diff
@@ -123,13 +124,10 @@ const JsonInput = props => {
   const editor = editorType === JSONKEY ? getJsonEditor() : getNormalEditor();
 
   const toggleIcon = (
-    <i
+    <Icon
```
[the remark this answers is not in the document]
Comment (lastnamearya) [reply console/src/components/Common/CustomInputTypes/JsonInput.js]: Fontawesome changed this icon in its latest version 5.0, I checked this before integrating. Now reverting back to the old icon.

[the remark this answers is not in the document]
Comment (lastnamearya) [reply console/src/components/Common/CustomInputTypes/JsonInput.js]: alternative looks like weird and out of shape, can we use this icon for expand @beerose ?

[the remark this answers is not in the document]
Comment (lastnamearya) [reply console/src/components/Common/CustomInputTypes/JsonInput.js]: Screenshot 2020-03-25 at 9 34 05 PM

during the all-hands call on Zoom, I find ~ Zoom is also using the same icon 😸

[the remark this answers is not in the document]
Comment (lastnamearya) [reply console/src/components/Common/CustomInputTypes/JsonInput.js]: reference ~ React Icons https://react-icons.netlify.com/#/icons/fa

**Comment by lastnamearya** [comment]
Screenshot 2020-03-28 at 1 27 20 AM
Screenshot 2020-03-28 at 1 24 44 AM

table icon fixed.

**Comment by lastnamearya** [comment]
Screenshot 2020-03-28 at 1 36 46 AM

Metadata status icon fixed.

File: console/src/components/Common/Common.scss [hunk 3]
```diff
@@ -181,13 +181,6 @@ table tbody tr th {
   display: inline-block;
 }
 
-.fontAwosomeClose {
```
[the remark this answers is not in the document]
Comment (beerose) [reply console/src/components/Common/Common.scss]: This class name is still used in the above files.

File: console/src/components/Services/Actions/Common/components/TypeDefinitionEditor.js [hunk 4]
```diff
@@ -75,16 +75,15 @@ const ActionDefinitionEditor = ({
...
+              <Icon mr="xs" type="close" color="red.primary" size={20} />
               <div>{errorMessage}</div>
             </div>
           )}
           <a
             className={`${styles.cloneTypeText} ${styles.cursorPointer} ${styles.add_mar_right}`}
             onClick={toggleModal}
           >
-            <CopyIcon className={styles.add_mar_right_small} />
-            Clone an existing type
```
Comment (beerose) [inline console/src/components/Services/Actions/Common/components/TypeDefinitionEditor.js]: Bring back `Clone an existing type`.

File: console/src/components/UIKit/atoms/Icon/Icon.js [hunk 5]
```diff
@@ -2,6 +2,8 @@ import styled from 'styled-components';
 import { color, typography, layout, space } from 'styled-system';
 
 export const StyledIcon = styled.svg`
+  cursor: ${({ cursor }) => (cursor ? 'pointer' : '')};
```
Comment (beerose) [inline console/src/components/UIKit/atoms/Icon/Icon.js]: This solution limits setting cursor only to `pointer`, but there are also other types (37 if I counted correctly). 

Also, the name `cursor` doesn't suggest what value should be set for `cursor` property and the `Icon` component silently assumes it's `pointer`. 

Possible solutions:
1. Add the possibility to explicitly set `cursor=pointer` for component `Icon`.
2. Simpler, but still limited solution — rename prop name to `pointer`.

File: console/src/components/Services/Actions/Common/components/TypeDefinitionEditor.js (hunk 4 above)
Comment (lastnamearya) [reply console/src/components/Services/Actions/Common/components/TypeDefinitionEditor.js]: fixed.

File: console/src/components/UIKit/atoms/Icon/Icon.js (hunk 5 above)
Comment (lastnamearya) [reply console/src/components/UIKit/atoms/Icon/Icon.js]: I renamed it to pointer currently.

**Comment by beerose** [comment]
We'll tackle the design system integration as a part of bigger design changes.

**Pull request closed without merging**